packages/client/src/pages/MainPage/ui/MainHeader/ui/MainHeader.tsx

```diff
@@ -3,16 +3,44 @@ import {
   HealthState,
   useNotifications,
 } from '@/app/providers/NotificationsProvider';
-import { useAppSelector } from '@/shared/lib/redux';
+import { LeaderboardConfig } from '@/pages/LeaderboardPage/ui/LeaderboardPage';
+import { LeaderboardAPI, LeaderboardData } from '@/shared/lib/api';
+import { useAppDispatch, useAppSelector } from '@/shared/lib/redux';
 import { LevelOrHpBar } from '@/shared/ui/LevelOrHpBar';
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 import css from './MainHeader.module.scss';
+import { addLeaderboardLevel } from '@/entities/tamagotchi';
 
 export const MainHeader = () => {
   const { expMax, expCurrent, hpCurrent, hpMax, level } = useAppSelector(
     (state) => state.tamagotchi,
   );
 
+  const { id } = useAppSelector((state) => state.user);
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    if (level === 1) {
+      LeaderboardAPI.all(LeaderboardConfig)
```

Comment:
Давай сделаем вложенность меньше, улучшим читаемость и добавим early return:
```suggestion
    if (level !== 1) return
    // тут остальной код
```

Reply:
Поправила